apps/aecontract/src/aect_create_tx.erl

```diff
@@ -128,13 +128,26 @@ nonce(#contract_create_tx{nonce = Nonce}) ->
 origin(#contract_create_tx{owner = OwnerPubKey}) ->
     OwnerPubKey.
 
-%% Owner should exist, and have enough funds for the fee
--spec check(tx(), aetx:tx_context(), aec_trees:trees(), height(), non_neg_integer()) -> {ok, aec_trees:trees()} | {error, term()}.
-check(#contract_create_tx{owner = OwnerPubKey, nonce = Nonce,
+%% Owner should exist, and have enough funds for the fee, the amount and the gas
+-spec check(tx(), aetx:tx_context(), aec_trees:trees(), height(), non_neg_integer()) ->
+                   {ok, aec_trees:trees()} | {error, term()}.
+check(#contract_create_tx{owner = OwnerPubKey,
+                          nonce = Nonce,
+                          amount     = Amount,
+                          gas        = Gas,
+                          gas_price  = GasPrice,
                           fee = Fee}, _Context, Trees, Height, _ConsensusVersion) ->
+    TotalAmount = Fee + Amount + Gas * GasPrice,
     Checks =
-        [fun() -> aetx_utils:check_account(OwnerPubKey, Trees, Height, Nonce, Fee) end],
-
+        [fun() -> aetx_utils:check_account(OwnerPubKey, Trees, Height, Nonce, TotalAmount) end,
+         %% TODO: Check minum gas price.
+         fun () ->
+                 case Fee >= aec_governance:minimum_tx_fee() of
```

Comment:
Is this not already checked by [tx generic code](https://github.com/aeternity/epoch/blob/10b0a4e9a03d178b0d6fed470d7b246b2ae876c0/apps/aecore/src/aec_trees.erl#L191)?

Reply:
It is.